src/App.tsx

```diff
@@ -1,21 +1,44 @@
 import "./App.css";
-import React, { useState } from "react";
-import { taskList } from "./taskList";
-
+import React, { useState, useEffect } from "react";
 import icon from "./icons8-delete.svg";
 
-
-//
 function TaskShow() {
-// Create a list with additional field to mark deleted task
-  const [tasks, setTasks] = useState(
-    taskList.map((task) => ({
-      id: task.id,
-      name: task.name,
-      isDeleted: false,
-    }))
-  );
-  const [newTaskName, setNewTaskName] = useState("");
+
+  interface Item {
+    id: number;
+    name: string;
+  }
+  interface ExItem {
+    id: number;
+    name: string;
+    isDeleted: boolean;
+  }
+  const [newTaskName, setNewTaskName] = useState("");  
+  //    Fetch data
+    const [tasks, setTasks] = useState<ExItem[]>([]);
+    
+    useEffect(() => {
+      fetch("https://www.jsonkeeper.com/b/MF8H")
+        .then((response) => {
+          if (!response.ok) {
+            throw new Error(response.statusText);
+          }
+          return response;
```

Comment:
we can also do `return response.json()` here instead of adding another `.then` below